src/app/components/Staking/Staking.tsx

```diff
@@ -1,225 +1,239 @@
+import { signForm } from "@/utils/signForm";
+import { getStakingTerm } from "@/utils/getStakingTerm";
+import { FinalityProviders } from "./FinalityProviders/FinalityProviders";
+import { WalletProvider } from "@/utils/wallet/wallet_provider";
+import { stakingSignReady } from "@/utils/stakingSignReady";
+import { GlobalParamsVersion } from "@/app/types/globalParams";
+import { Delegation } from "@/app/types/delegations";
+import { Loading } from "./Loading";
+import { WalletNotConnected } from "./Form/States/WalletNotConnected";
+import { Message } from "./Form/States/Message";
+import { StakingTime } from "./Form/StakingTime";
+import { StakingAmount } from "./Form/StakingAmount";
 import { PreviewModal } from "../Modals/PreviewModal";
-import { blocksToTime } from "@/utils/blocksToTime";
-import { ConnectLarge } from "../Connect/ConnectLarge";
-import { satoshiToBtc, btcToSatoshi } from "@/utils/btcConversions";
+import stakingCapReached from "./Form/States/staking-cap-reached.svg";
+import stakingNotStarted from "./Form/States/staking-not-started.svg";
+import stakingUpgrading from "./Form/States/staking-upgrading.svg";
 
-interface StakingParams {
-  minStakingAmountSat: number;
-  maxStakingAmountSat: number;
-  minStakingTime: number;
-  maxStakingTime: number;
-  stakingCapSat: number;
-}
+const stakingFeeSat = 500;
 
 interface StakingProps {
-  amountSat: number;
-  onAmountSatChange: (amountSat: number) => void;
-  term: number;
-  onTermChange: (term: number) => void;
   finalityProviders: FinalityProviderInterface[] | undefined;
-  selectedFinalityProvider: FinalityProviderInterface | undefined;
-  // called when the user selects a finality provider
-  onFinalityProviderChange: (btcPkHex: string) => void;
-  onSign: () => void;
-  stakingParams: StakingParams | undefined;
   isWalletConnected: boolean;
-  overTheCap: boolean;
   isLoading: boolean;
-  isUpgrading: boolean | undefined;
-  // if the staking cap is reached, the user can't stake
+  overTheCap: boolean;
   onConnect: () => void;
+  btcWallet: WalletProvider | undefined;
+  btcWalletNetwork: networks.Network | undefined;
+  address: string | undefined;
+  publicKeyNoCoord: string;
+  setDelegationsLocalStorage: Dispatch<SetStateAction<Delegation[]>>;
+  paramWithContext:
+    | {
+        currentVersion: GlobalParamsVersion | undefined;
+        isApprochingNextVersion: boolean | undefined;
+      }
+    | undefined;
 }
 
 export const Staking: React.FC<StakingProps> = ({
-  amountSat,
-  onAmountSatChange,
-  term,
-  onTermChange,
   finalityProviders,
-  selectedFinalityProvider,
-  onFinalityProviderChange,
-  onSign,
-  stakingParams,
   isWalletConnected,
+  isLoading,
   overTheCap,
   onConnect,
-  isLoading,
-  isUpgrading,
+  paramWithContext,
+  btcWallet,
+  btcWalletNetwork,
+  address,
+  publicKeyNoCoord,
+  setDelegationsLocalStorage,
 }) => {
+  // Staking form state
+  const [stakingAmountBTC, setStakingAmountBTC] = useState(0);
```

Comment:
```suggestion
  const [stakingAmountSat, setStakingAmountSat] = useState(0);
```

We should always use satoshis internally.

Reply:
Done